src/Components/RecsListTable/index.js

```diff
@@ -63,117 +72,129 @@ const RecsListTable = () => {
+              ),
+            },
+            {
+              title: (
+                <div key={key}>
+                  <Tooltip
+                    key={key}
+                    position={TooltipPosition.bottom}
+                    content={intl.formatMessage(
+                      messages.rulesDetailsTotalRiskBody,
+                      {
+                        risk:
+                          TOTAL_RISK_LABEL_LOWER[value.total_risk] ||
+                          intl.formatMessage(messages.undefined),
+                        strong: (str) => strong(str),
+                      }
+                    )}
+                  >
+                    {value?.total_risk ? (
+                      <InsightsLabel value={value.total_risk} />
+                    ) : (
+                      intl.formatMessage(messages.nA)
+                    )}
+                  </Tooltip>
+                </div>
+              ),
+            },
+            {
+              title: (
+                <div key={key}>{`${
+                  value?.impacted_clusters_count !== undefined
+                    ? value.impacted_clusters_count.toLocaleString()
+                    : intl.formatMessage(messages.nA)
+                }`}</div>
+              ),
+            },
+          ],
+        },
+        {
+          fullWidth: true,
+          cells: [
+            {
+              title: (
+                <section className="pf-m-light pf-l-page__main-section pf-c-page__main-section">
+                  <Stack hasGutter>
+                    <RuleDetails
+                      rule={{
+                        ...value,
+                        impact: { impact: value.impact },
+                        // TODO: fix <Router> issue in the async component and then remove the line below
+                        impacted_clusters_count: undefined,
+                      }}
+                      isDetailsPage={false}
+                    />
+                  </Stack>
+                </section>
+              ),
+            },
+          ],
+        },
+      ]);
   };
 
-  const buildDisplayedRows = (rows) => {
+  const buildDisplayedRows = (rows, index, direction) => {
+    const sortedRecommendations = [
+      'description',
+      'publish_date',
+      'total_risk',
+      'impacted_clusters_count',
+    ];
     return rows
+      .sort((firstItem, secondItem) =>
+        firstItem[sortedRecommendations[index]] >
+        secondItem[sortedRecommendations[index]]
+          ? 1
+          : secondItem[sortedRecommendations[index]] >
+            firstItem[sortedRecommendations[index]]
+          ? -1
+          : 0
+      )
       .slice(
         filters.limit * (page - 1),
         filters.limit * (page - 1) + filters.limit
       )
       .flatMap((row, index) => {
         const updatedRow = [...row];
         row[1].parent = index * 2;
+        direction === SortByDirection.asc ? updatedRow : updatedRow.reverse();
```

Comment:
Why you would ever change the `direction` **parameter** of the `buildDisplayedRows` function?..

What is the point of reversing one single row? Modifying flatMap here is most probably not required at all.

Reply:
From what I can see, I would call reverse function right after `.sort` https://github.com/RedHatInsights/ocp-advisor-frontend/pull/56/files#diff-4a5517c05a9afe6cc744c7b111538de5db09e9c5fd1a2aac0d315dd51a1f1da4R181